ingest/ingest.py

```diff
@@ -0,0 +1,451 @@
+#!/usr/bin/env python
+"""
+Script to transfer 1kg data and populate the thousand-genomes project
+"""
+import subprocess
+from os.path import basename, join, exists, dirname
+from typing import Dict, Union, List, Optional
+import logging
+
+import pandas as pd
+import click
+
+from cpg_production_pipelines.pipeline import setup_batch
+from cpg_production_pipelines import utils
+from sample_metadata.parser.generic_parser import GenericParser, GroupedRow
+
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(format='%(levelname)s (%(name)s %(lineno)s): %(message)s')
+logger.setLevel(logging.INFO)
+
+
+METADATA_XLSX_URL = (
+    'ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/'
+    '20130606_sample_info/20130606_sample_info.xlsx'
+)
+PED_URL = (
+    'ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/'
+    '20121016_updated_pedigree/G1K_samples_20111130.ped'
+)
+SOURCE_BUCKET = 'gs://fc-56ac46ea-efc4-4683-b6d5-6d95bed41c5e'
+
+
+class TGParser(GenericParser):  # pylint: disable=too-many-instance-attributes
+    """
+    Inherits from sample_metadata's GenericParser class and implements parsing logic specific
+    to thousand-genomes
+    """
+
+    def __init__(
+        self,
+        sample_to_copy_n: Optional[int],
+        transfer_cram: bool,
+        transfer_gvcf: bool,
+        namespace: str,
+        overwrite: bool,
+        sample_metadata_project: str,
+        use_batch: bool,
+    ):
+        # Base bucket to copy files to
+        self.target_bucket = f'gs://cpg-{sample_metadata_project}-{namespace}-upload'
+        super().__init__(self.target_bucket, sample_metadata_project)
+        self.samples_to_copy_n = sample_to_copy_n
+        self.transfer_cram = transfer_cram
+        self.transfer_gvcf = transfer_gvcf
+        self.overwrite = overwrite
+        self.resources_dir = 'resources'  # for downloaded files and bucket lists
+        self.tmp_dir = 'tmp'  # temporary dir for intemediate files
+        if use_batch:
+            self.batch = setup_batch(
+                title='Transfer 1KG data',
+                keep_scratch=False,
+                tmp_bucket=f'cpg-{sample_metadata_project}-{namespace}-tmp',
+                analysis_project=sample_metadata_project,
+            )
+        else:
+            self.batch = None
+        _call(f'mkdir -p {self.tmp_dir}')
+        _call(f'mkdir -p {self.resources_dir}')
+
+    def get_sample_id(self, row: Dict[str, any]) -> str:
+        return row['s']
+
+    def get_sample_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, any]:
+        sample_meta = dict()
+        sample_meta['continental_pop'] = row['continental_pop']
+        sample_meta['fam_id'] = row['fam_id']
+        sample_meta['pat_id'] = row['pat_id']
+        sample_meta['mat_id'] = row['mat_id']
+        sample_meta['sex'] = row['sex']
```

Comment:
Can we call this `gender` to align with what they call it from upstream? Otherwise is it possible to list a source saying that "what they call gender is actually reported sex"?

I intend to refactor the sex stuff imminently based on feedback the other week.